packages/gatsby-plugin-page-creator/src/is-valid-collection-path-implementation.ts

```diff
@@ -18,12 +19,26 @@ export function isValidCollectionPathImplementation(
   parts.forEach(part => {
     if (!part.includes(`{`) && !part.includes(`}`)) return
 
-    const model = part.match(/\{([a-zA-Z_]\w*)./)?.[1]! // Search for word before first dot, e.g. Model
-    const field = part.match(/((?<=\.).*)}/)?.[1]! // Search for everything after the first dot, e.g. foo__bar (or in invalid case: foo.bar)
+    const model = matchAllPolyfill(/\{([a-zA-Z_]\w*)./g, part) // Search for word before first dot, e.g. Model
+    const field = matchAllPolyfill(/{.*?((?<=\w\.)[^}]*)}/g, part) // Search for everything after the first dot, e.g. foo__bar (or in invalid case: foo.bar)
```

Comment:
Should the search of the second star be greedy or non-greedy?

Consider these two regexes
```js
/{.*?((?<=\w\.)[^}]*)}/g
/{.*?((?<=\w\.)[^}]*?)}/g
```